DQM/SiTrackerPhase2/plugins/Phase2ITMonitorCluster.cc

```diff
@@ -0,0 +1,416 @@
+    //MonitorElement* XYGlobalPositionMapPixel=nullptr;
+    MonitorElement* XYLocalPositionMapPixel = nullptr;
+  };
+
+  MonitorElement* numberClusters_;
+  MonitorElement* globalXY_barrel_;
+  MonitorElement* globalXY_endcap_;
+  MonitorElement* globalRZ_barrel_;
+  MonitorElement* globalRZ_endcap_;
+
+  void bookLayerHistos(DQMStore::IBooker& ibooker, uint32_t det_it, std::string& subdir);
+
+  std::map<std::string, ClusterMEs> layerMEs_;
+  edm::ParameterSet config_;
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> itPixelClusterToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoToken_;
+  const TrackerGeometry* tkGeom_ = nullptr;
+  const TrackerTopology* tTopo_ = nullptr;
+};
+#endif
+
+#include "DQM/SiTrackerPhase2/interface/TrackerPhase2DQMUtil.h"
+
+//
+// constructors
+//
+Phase2ITMonitorCluster::Phase2ITMonitorCluster(const edm::ParameterSet& iConfig)
+    : config_(iConfig),
+      itPixelClusterToken_(consumes<edmNew::DetSetVector<SiPixelCluster>>(
+          config_.getParameter<edm::InputTag>("InnerPixelClusterSource"))),
+      geomToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord, edm::Transition::BeginRun>()),
+      topoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd, edm::Transition::BeginRun>()) {
+  edm::LogInfo("Phase2ITMonitorCluster") << ">>> Construct Phase2ITMonitorCluster ";
+}
+
+Phase2ITMonitorCluster::~Phase2ITMonitorCluster() {
+  edm::LogInfo("Phase2ITMonitorCluster") << ">>> Destroy Phase2ITMonitorCluster ";
+}
+//
+// -- DQM Begin Run
+void Phase2ITMonitorCluster::dqmBeginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) {
+  edm::ESHandle<TrackerGeometry> geomHandle = iSetup.getHandle(geomToken_);
+  tkGeom_ = &(*geomHandle);
+  edm::ESHandle<TrackerTopology> tTopoHandle = iSetup.getHandle(topoToken_);
+  tTopo_ = tTopoHandle.product();
+}
+
+//
+// -- Analyze
+//
+void Phase2ITMonitorCluster::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  // Getting the clusters
+  edm::Handle<edmNew::DetSetVector<SiPixelCluster>> itPixelClusterHandle;
+  iEvent.getByToken(itPixelClusterToken_, itPixelClusterHandle);
+  // Number of clusters
+  std::map<std::string, unsigned int> nClsmap;
+  unsigned int nclusGlobal = 0;
+  for (const auto& DSVItr : *itPixelClusterHandle) {
+    uint32_t rawid(DSVItr.detId());
+    DetId detId(rawid);
+    const GeomDet* geomDet = tkGeom_->idToDet(detId);
+    if (!geomDet)
+      continue;
+    const GeomDetUnit* geomDetUnit(tkGeom_->idToDetUnit(detId));
+    if (!geomDetUnit)
+      continue;
+    nclusGlobal += DSVItr.size();
+    std::string folderkey = phase2tkutil::getITHistoId(detId, tTopo_);
+    // initialize the nhit counters if they don't exist for this layer
+    if (nClsmap.find(folderkey) == nClsmap.end()) {
+      nClsmap.emplace(folderkey, 0);
+    }
+    nClsmap.at(folderkey) += DSVItr.size();
```

Comment:
you might use the outcome of find to avoid searching the map twice

Reply:
@davidlange6 thanks! are you recommending the following? 

   auto counterDet = nClsmap.find(folderkey);
   if (counterDet == nClsmap.end()) {
     nClsmap.emplace(folderkey, DSVItr.size());
   } else counterDet->second += DSVItr.size();

Reply:
That looks right.